src/Autocomplete/tests/Functional/AutocompleteFormRenderingTest.php

```diff
@@ -123,4 +123,55 @@ public function testMultipleDoesNotFailWithoutSelectedChoices()
             ->assertElementCount('#product_ingredients option', 0)
         ;
     }
+
+    public function testItUsesPassedExtraOptions()
+    {
+        $ingredient1 = IngredientFactory::createOne(['name' => 'Flour']);
+        $ingredient2 = IngredientFactory::createOne(['name' => 'Sugar']);
+        $ingredient3 = IngredientFactory::createOne(['name' => 'Modified Flour']);
+
+        $this->browser()
+            ->throwExceptions()
+            ->get('/test-form')
+            ->assertElementCount('#product_ingredients option', 0)
+            ->assertNotContains('Flour')
+            ->assertNotContains('Sugar')
+            ->assertNotContains('Modified Flour')
+            // request all three ingredients
+            ->post('/test-form', [
+                'body' => [
+                    'product' => [
+                        'ingredients' => [
+                            (string) $ingredient1->getId(),
+                            (string) $ingredient2->getId(),
+                            (string) $ingredient3->getId(),
+                        ],
+                    ],
+                ],
+            ])
+            // assert that "Modified Flour" is not included
+            ->assertElementCount('#product_ingredients option', 2)
+            ->assertContains('Flour')
+            ->assertContains('Sugar')
+            ->assertNotContains('Modified Flour')
+        ;
+    }
+
+    public function testItReturnsErrorWhenSendingMalformedExtraOptions(): void
```

Comment:
This test does not provide much for me, as there is working comparaison.. 

I mean instead of the "$this->encodeData" you could pass "foo" to get the same result no ?

Reply:
No, I wanted to check two cases here:
1. `@checksum` is required
2. `@checksum` must be valid
I could add also 3rd case `$extraOptionsWithValidChecksum` and check if the status is 200 🤔. I'll add it, and then the test should be complete and sensible.

Reply:
Perfect

Reply:
Done, it should be better now :D.